### PR TITLE
CASSANDRA-17706 Make sure existing delayed tasks in StreamTransferTask cannot prevent clean shutdown

### PR DESCRIPTION
**Pull request by maedhroz**

### CODE REVIEW

**Comment by maedhroz** [comment]
https://app.circleci.com/pipelines/github/maedhroz/cassandra?branch=CASSANDRA-17706-4.0&filter=all

File: src/java/org/apache/cassandra/streaming/StreamTransferTask.java [hunk 1]
```diff
@@ -62,6 +61,7 @@ public class StreamTransferTask extends StreamTask
     public StreamTransferTask(StreamSession session, TableId tableId)
     {
         super(session, tableId);
+        timeoutExecutor.setExecuteExistingDelayedTasksAfterShutdownPolicy(false);
```
Comment (maedhroz) [inline src/java/org/apache/cassandra/streaming/StreamTransferTask.java]: @dcapwell My thinking here is that given no non-test codepath actually shuts `timeoutExecutor` down, this should be a safe change.

Comment (maedhroz) [reply src/java/org/apache/cassandra/streaming/StreamTransferTask.java]: Also, when this timeout is scheduled in prod code...

```
task.scheduleTimeout(message.header.sequenceNumber, 12, TimeUnit.HOURS);
```

:D

**Review by dcapwell — APPROVED** [review]

**Comment by maedhroz** [comment]
Committed as https://github.com/apache/cassandra/commit/924cd8f52c2437d0a0ce8c48196bb6aa8976fbb8

**Pull request closed without merging**